src/DataStructures/Tensor/Expressions/Evaluate.hpp

```diff
@@ -8,35 +8,115 @@
 
 #include "DataStructures/Tensor/Expressions/TensorExpression.hpp"
 #include "DataStructures/Tensor/Tensor.hpp"
+#include "Utilities/Algorithm.hpp"
 #include "Utilities/Requires.hpp"
 
 namespace TensorExpressions {
+/*!
+ * \ingroup TensorExpressionsGroup
+ * \brief Determines and stores a LHS tensor's symmetry and index list from a
+ * RHS tensor expression and desired LHS index order
+ *
+ * \details Given the generic index order of a RHS TensorExpression and the
+ * generic index order of the desired LHS Tensor, this creates a mapping between
+ * the two that is used to determine the (potentially reordered) ordering of the
+ * elements of the desired LHS Tensor`s ::Symmetry and typelist of
+ * \ref SpacetimeIndex "TensorIndexType"s.
+ *
+ * @tparam RhsTensorIndexList the typelist of TensorIndex of the RHS
+ * TensorExpression, e.g. `ti_a_t`, `ti_b_t`, `ti_c_t`
+ * @tparam LhsTensorIndexList the typelist of TensorIndexs of the desired LHS
+ * tensor, e.g. `ti_b_t`, `ti_c_t`, `ti_a_t`
+ * @tparam RhsSymmetry the ::Symmetry of the RHS indices
+ * @tparam RhsTensorIndexTypeList the RHS TensorExpression's typelist of
+ * \ref SpacetimeIndex "TensorIndexType"s
+ */
+template <typename RhsTensorIndexList, typename LhsTensorIndexList,
+          typename RhsSymmetry, typename RhsTensorIndexTypeList,
+          size_t NumIndices = tmpl::size<RhsSymmetry>::value,
+          typename IndexSequence = std::make_index_sequence<NumIndices>>
+struct LhsTensor;
```

Comment:
I found it slightly confusing that this is called `LhsTensor` because if I correctly understand the documentation and code, this isn't a tensor so much as it is a mapping between tensor indices. If that's correct, I'd suggest a rename to make the functionality more evident, perhaps something like `LhsTensorIndexMap` or `TensorIndexMap` or ...? Feel free to throw out other suggestions too if you agree with the general idea but don't like these particular names.

Reply:
Thanks! I had a similar reservation when I was deciding on the name. I'll rename!

Reply:
I think I see what you're saying, and I agree my first suggestions aren't ideal. Here's some other options I'll throw out: `LhsTensorIndexHelper` which is in the same spirit as your `LhsTensorIndexInfo` but, personally, I like it a little better; `LhsTensorIndices` (but of course there's also the symmetry information); `LhsTensorSymmAndIndices` (maybe my favorite to far?)... do any of these appeal to you?

I guess if nothing sticks at this stage, I'll suggest you keep the name as it currently is and we don't waste too much of your time on this 🙂

Reply:
I think `LhsTensorSymmAndIndices` is the most descriptive, so I went with that!